OrbitGl/GlCanvas.cpp

```diff
@@ -474,3 +480,20 @@ int GlCanvas::AddText2D(const char* a_Text, int a_X, int a_Y, float a_Z,
   return m_TextRenderer.AddText2D(a_Text, a_X, a_Y, a_Z, a_Color, a_MaxSize,
                                   a_RightJustified, a_InvertY);
 }
+
+void GlCanvas::ResetHoverTimer() {
+  m_HoverTimer.Reset();
+  m_CanHover = true;
+}
+
+[[nodiscard]] PickingMode GlCanvas::GetPickingMode() {
+  PickingMode picking_mode = PickingMode::kNone;
+  if (m_Picking) {
+    picking_mode = PickingMode::kClick;
+  }
+  if (m_IsHovering) {
+    picking_mode = PickingMode::kHover;
```

Comment:
I am not sure if this is intended - if case of m_IsHovering and m_Picking this will override PickingMode::kClick.

Reply:
Good point... it's like this for a while now, and it will only appear if hover and click trigger in the exact same rendering frame, which is super unlikely. But you're right, I've changed it as suggested in the previous comment, which gets rid of this issue as well.

Reply:
I've just realized that this breaks hovering, because it actually was intended this way - m_Picking is set on click and on hover. We should change this to be more precise, like distinguish between click and hover or something. I've changed the check now to retain the old behavior, should refactor this in the future.